test/spatial_reaction_systems/lattice_reaction_systems_jumps.jl

```diff
@@ -0,0 +1,309 @@
+    pD_1 = [:dI => 0.02, :dS => 0.01, :dR => 0.03]
+    pD_2 = [:dI => 0.02, :dS => fill(0.01, ne(small_2d_grid)), :dR => 0.03]
+    pD_3 = [0.01, 0.02, 0.03]
+    pD_4 = [fill(0.01, ne(small_2d_grid)), 0.02, 0.03]
+    pD_5 = permutedims(hcat(fill(0.01, ne(small_2d_grid)), fill(0.02, ne(small_2d_grid)), fill(0.03, ne(small_2d_grid))))
+
+    # Checks hopping rates and u0 are correct.
+    true_u0 = [fill(1.0, 1, 25); fill(2.0, 1, 25); fill(3.0, 1, 25)]
+    true_hopping_rates = cumsum.([fill(dval, length(v)) for dval in [0.01,0.02,0.03], v in small_2d_grid.fadjlist])
+    for u0 in [u0_1, u0_2, u0_3, u0_4, u0_5]
+        # Provides parameters as a tupple.
+        for pC in [pC_1, pC_3], pD in [pD_1, pD_2, pD_3, pD_4, pD_5]
+            dprob = DiscreteProblem(lrs, u0, (0.0, 100.0), (pC,pD))
+            jprob = JumpProblem(lrs, dprob, NSM())
+            @test jprob.prob.u0 == true_u0
+            @test jprob.discrete_jump_aggregation.hop_rates.hop_const_cumulative_sums == true_hopping_rates
+        end
+        # Provides parameters as a combined vector.
+        for pC in [pC_1], pD in [pD_1, pD_2]
+            dprob = DiscreteProblem(lrs, u0, (0.0, 100.0), [pD; pC])
+            jprob = JumpProblem(lrs, dprob, NSM())
+            @test jprob.prob.u0 == true_u0
+            @test jprob.discrete_jump_aggregation.hop_rates.hop_const_cumulative_sums == true_hopping_rates
+        end
+    end
+end
+
+### Hoping Rates Computation Tests ###
+
+# Currently not in use, but will be added as more cases are enabled.
+if false
+    # Tests hopping rates of the form D_{s,i,j}
+    let
+        # Prepares the system.
+        lrs = LatticeReactionSystem(SIR_system, SIR_srs_2, small_2d_grid)
+        u0 = [:I => 2.0, :S => 1.0, :R => 3.0]
+        pC = [:β => 0.2, :α => 0.1]
+
+        # Prepare various (diffusion) parameter input types.
+        pD_1 = [:dI => 0.02, :dS => 0.01, :dR => 0.03]
+        pD_2 = [:dI => 0.02, :dS => fill(0.01, ne(lrs.lattice)), :dR => 0.03]
+        pD_3 = [0.01, 0.02, 0.03]
+        pD_4 = [fill(0.01, ne(lrs.lattice)), 0.02, 0.03]
+        pD_5 = permutedims(hcat(fill(0.01, ne(lrs.lattice)), fill(0.02, ne(lrs.lattice)), fill(0.03, ne(lrs.lattice))))
+
+        # Checks hopping rates are correct.
+        true_hopping_rates = []
+        # Provides parameters as a tupple.
+        for pD in [pD_1, pD_2, pD_3, pD_4, pD_5]
+            dprob = DiscreteProblem(lrs, u0, (0.0, 100.0), (pC,pD))
+            jprob = JumpProblem(lrs, dprob, NSM())
+            @test jprob.discrete_jump_aggregation.hop_rates.hop_const_cumulative_sums == true_hopping_rates
+        end
+        # Provides parameters as a combined vector.
+        for pD in [pD_1, pD_2]
+            dprob = DiscreteProblem(lrs, u0, (0.0, 100.0), [pD; pC])
+            jprob = JumpProblem(lrs, dprob, NSM())
+            @test jprob.discrete_jump_aggregation.hop_rates.hop_const_cumulative_sums == true_hopping_rates
+        end
+    end
+
```

Comment:
I recommend taking out these `if else` codeblocks. There is no reason to check them in if they won't actually run. We'll add them in subsequent PRs to keep this one smaller.

```suggestion

```

Reply:
Dead code has been removed.